services/ui/ws/window_server.cc

```diff
@@ -751,9 +751,25 @@ void WindowServer::OnWindowVisibilityChanged(ServerWindow* window) {
 
   WindowManagerDisplayRoot* display_root =
       display_manager_->GetWindowManagerDisplayRoot(window);
-  if (display_root)
+  if (display_root) {
     display_root->window_manager_state()->ReleaseCaptureBlockedByModalWindow(
         window);
+
+    if (IsInExternalWindowMode() && window->visible()) {
+      display_root->display()->platform_display()->Show();
+    }
+  }
+}
+
+void WindowServer::OnHideNativeWindow(ServerWindow* window) {
```

Comment:
Nope, if we hide window on each visibilty changed to false, main window will just disappear. That's the reason why I have it separately from visibility.